### PR TITLE
Redis: reenable support connecting to servers using TLS with self-signed certificates (e.g., Heroku Redis)

### PR DESCRIPTION
**Pull request by dbertram**
**tl;dr:** This PR re-enables support for connecting to Redis servers using TLS with self-signed certificates. It also updates W3TC's "cache plugin config in redis" logic to support Redis servers with self-signed certificates. This updated implementation should now play nicely with older versions of `phpredis` and avoid generating [warnings about passing the wrong # of args to `pconnect`](https://github.com/W3EDGE/w3-total-cache/issues/533).

@KZeni If you're up for it, I'd love to have you test out this updated code with your setup that's using `phpredis 3.1.1`!

## Timeline:
1. [My previous PR](https://github.com/W3EDGE/w3-total-cache/pull/494) to add support for connecting to redis using TLS with self-signed certs (e.g., Heroku Redis) was accepted on April 28, 2022 ✅ 
2. Support for disabling TLS cert verification ships with W3TC v2.2.2 🎉 
3. It was removed in v2.2.3 https://github.com/W3EDGE/w3-total-cache/pull/526
   - some folks with older `phpredis` were getting [warnings about passing the wrong # of args to pconnect](https://github.com/W3EDGE/w3-total-cache/issues/533), so support for disabling TLS cert verification was removed + the corresponding plugin config UI for the option was hidden
4. You are here :)

This PR may be easier to review as a set of individual commits. I tried to make each commit stand on its own.

## This PR:
- e2482c778b87700fbb84afb48eb6f14b2a4ef832 Cleans up an extraneous call to `restore_error_handler` left over from a release candidate
- 0ffd86b521601915cd4563fc4786d491b4a1cc97 DRYs increasingly confusing logic for determining the redis connection args
- 874732b8465797685545a646340855aa15205bc3 Improves logic related to the `read_timeout` connection arg (it was added to `phpredis` in v3.1.3 not v5)
- a076c93187a726d90778960b6a4318093cddb5b0 **Re-enables support for disabling TLS cert verification**
- 238df91c86d37825c8709ed42da558c3b69fac73 Adds ability to disable TLS cert verification for the "cache plugin config in redis" logic
- 20ce726e56e6b089570ea7ec642d3aca7d87a5b4 Fixes up some odds and ends that use redis, but weren't respecting the "verify_tls_certificates" option

Related to #67

### CODE REVIEW

**Comment by KZeni** [comment]
Okay, I did need to edit `w3-total-cache/Root_Loader.php` on top of this PR for improved PHP version compatibility (otherwise gave a site-wide fatal PHP error) where:
```
                if ( 'never' === $visibility || ( 'extension' === $visibility && ! isset( $extensions['imageservice'] ) ) ) {
                        add_action(
                                'pre_get_posts',
                                array( $this, 'w3tc_modify_query_obj' ),
                        );

                        add_filter(
                                'ajax_query_attachments_args',
                                array( $this, 'w3tc_filter_ajax_args' ),
                        );
                }
```
needed to be:
```
                if ( 'never' === $visibility || ( 'extension' === $visibility && ! isset( $extensions['imageservice'] ) ) ) {
                        add_action(
                                'pre_get_posts',
                                array( $this, 'w3tc_modify_query_obj' )
                        );

                        add_filter(
                                'ajax_query_attachments_args',
                                array( $this, 'w3tc_filter_ajax_args' )
                        );
                }
```
*Notice the trailing comma at the end of the parameters of the add_filter function being called... it's unnecessary, and some PHP versions (ex. 7.2 & possibly others) don't like the extra comma when it comes to separating attributes being provided in a function call (to the point of a fatal server error being caused from it being there.)

**Comment by dbertram** [comment]
@KZeni Ah, interesting. Thanks for taking a look! It looks like those trailing commas were introduced in https://github.com/W3EDGE/w3-total-cache/pull/552.

@cssjoe Is that the kind of thing that should be fixed in its own PR? I'm happy to tack that fix on here if that makes the most sense.

@KZeni Did everything else in the PR work ok with your php/phpredis version? I.e., no return of https://github.com/W3EDGE/w3-total-cache/issues/533? 🤞

**Review by cssjoe — CHANGES_REQUESTED** [review]
@dbertram Thank you for your contribution.  The changes look ok but need further testing on our side.  We have the extra commas removed on our 2.2.5 patch release.  This PR does need some changes for https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/ -- remove the underscore on function names and adjust formatting.

**Comment by jacobd91** [comment]
[b80ef53](https://github.com/W3EDGE/w3-total-cache/pull/559/commits/b80ef53644f546a41a8e67ffe7201bdbcc8bd949) also includes a fix for missing p/connect call parameters for usage statistics

**Comment by dbertram** [comment]
@jacobd91 Thank you for hopping in on this PR and pushing those updates!

I had looked over [the contribution guidelines](https://github.com/W3EDGE/w3-total-cache/wiki/Contributor-Guidelines#code-style-rules), but failed to find any hints on the best/easiest way to get coding standards linting/formatting in place on my local clone. Is this documented somewhere and I just missed it?

**Comment by jacobd91** [comment]
Typically we use Visual Studio Code utilizing PHP_CodeSniffer version 3.7.0 (stable) by Squiz (http://www.squiz.net) with PEAR coding standard by default but have additional coding standards we test with. See list of used coding standards below:

`PEAR, WordPress-Extra, WordPress, WordPress-Docs and WordPress-Core`

Additionally we try our best to conform to the coding standards and best-practices documented on wordpress.org

**Review by cssjoe — APPROVED** [review]

**Comment by cssjoe** [comment]
@dbertram Thank you for your contribution.  The changes will be included in our next patch version.
